src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java

```diff
@@ -90,104 +109,110 @@ public void onApplicationEvent(ContextRefreshedEvent event) {
-					// time
-					List<StandardizationSettings> standardizationSettingses = StandardizationSettings
-							.findAllStandardizationSettingses();
-
-					if (standardizationSettingses.size() == 0) {
-						// If we have no standardization settings rows, create one and set the value to
-						// shouldStandardize to whatever the current settings say
-						// Note - this is a weird state, if we have previously applied standardized
-						// settings then we should have a standardization settings row
-						StandardizationSettings standardizationSettings = new StandardizationSettings();
-						standardizationSettings
-								.setNeedsStandardization(currentStandardizationSettings.getShouldStandardize());
-						standardizationSettings.setModifiedDate(new Date());
-						standardizationSettings.persist();
-					} else {
-
-						// If there is more than 0, then just update the current row
-						standardizationSettingses.get(0)
-								.setNeedsStandardization(currentStandardizationSettings.getShouldStandardize());
-						standardizationSettingses.get(0).setModifiedDate(new Date());
-						standardizationSettingses.get(0).merge();
-					}
-				} else {
-					logger.info("Standardizer configs have not changed, not marking 'standardization needed'");
-				}
-			} else {
-				logger.info("No standardization history found in database");
-				StandardizationSettings standardizationSettings = new StandardizationSettings();
-				standardizationSettings.setModifiedDate(new Date());
-				Long numberOfParents = Parent.countParents();
-				if (numberOfParents == 0.0) {
-					logger.info(
-							"There are no parents registered, we can assume standardization configs match the database standardization state so storing configs as the standardization settings");
-					standardizationSettings.setNeedsStandardization(false);
-					standardizationSettings.persist();
-					logger.info("Saved current standardization settings");
-					executeStandardization("acas", "Initialization");
+		// Get the current settings from the config file using "false" which tell getStandardizerSettings to get the raw configs as opposed to the fixed/applied configs
+		StandardizerSettingsConfigDTO currentRawStandardizerSettings = chemStructureService.getStandardizerSettings(false);
+
+		// Get the applied settings from the history table which should have the most recent standardization settings applied
+		List<StandardizationHistory> completedHistories = StandardizationHistory.findStandardizationHistoriesByStatus("complete", 0, 1, "id", "DESC").getResultList();
+
+		StandardizationHistory mostRecentHistory;
+		if (completedHistories.size() > 0) {
+			// Got a complete history where the standardization actually standaradized compounds
+			mostRecentHistory = completedHistories.get(0);
+		} else {
+			// There are 2 normal states where there are no completed histories
+			// 1. We upgraded to the new standardization code and there are no completed standardizations
```

Comment:
This comment block is useful, but I think it should be expanded a bit.
Specifically, explaining what will happen in this case 1, since it's not part of the `if` statement below. I'm thinking something like:
```
// In case 1, we pass along an empty StandardizationHistory, which will cause the system to report that it needs standardization.
// In case 2 we try to fill in an initial standardization record with the current settings, provided they're valid settings, so no restandardization is needed.
```

Reply:
Thanks, added those